sdk/tests/tutorials/ibor/test_reference_portfolios.py

```diff
@@ -33,16 +34,26 @@ def setUpClass(cls):
         cls.instrument_loader = InstrumentLoader(cls.instruments_api)
         cls.instrument_ids = cls.instrument_loader.load_instruments()
 
+        cls.id_generator = IdGenerator(default_scope=TestDataUtilities.tutorials_scope)
+
+    @classmethod
+    def tearDownClass(cls):
+        for _, scope, code in cls.id_generator.pop_scope_and_codes():
```

Comment:
I'm dreaming of this being further encapsulated/abstracted with something like cls.id_generator.delete_portfolios(portfolios_api), or cls.id_pop_scope_and_codes(api_factory, callable_method)

but perhaps this would be overengineering. Up to you Riz

Reply:
We could add another layer on top which wraps this, the complexity comes from the fact that that layer would need to know how to talk to LUSID and when it is safe to delete items. For now I think we keep it simple and if needed, we can add later

Reply:
That's fair. Can do that later when we decide to expand this Id Generator pattern to the rest of our SDKs